config-check/src/main/java/com/okta/commons/configcheck/ConfigurationValidator.java

```diff
@@ -42,75 +42,128 @@ private ConfigurationValidator() {}
      * </ul>
      * @param url The url to be validated
      */
-    public static void validateHttpsUrl(String url) {
-
-        hasText(url, ERRORS.getString("orgUrl.missing"));
-        doesNotContain(url, "{yourOktaDomain}", ERRORS.getString("orgUrl.containsBrackets"));
-        doesNotContain(url, "-admin.okta.com", formattedErrorMessage("orgUrl.containsAdmin", url));
-        doesNotContain(url, "-admin.oktapreview.com", formattedErrorMessage("orgUrl.containsAdmin", url));
-        doesNotContain(url, "-admin.okta-emea.com", formattedErrorMessage("orgUrl.containsAdmin", url));
-
-        try {
-            URL tempUrl = new URL(url);
-            if (!"https".equalsIgnoreCase(tempUrl.getProtocol())) {
-                throw new IllegalArgumentException(formattedErrorMessage("orgUrl.nonHttpsInvalid", url));
-            }
-            if (tempUrl.getHost().endsWith(".com.com")){
-                throw new IllegalArgumentException(formattedErrorMessage("orgUrl.invalid", url));
-            }
+    public static void assertOrgUrl(String url) {
+        validateOrgUrl(url).ifInvalidThrow();
+    }
 
-        } catch (MalformedURLException e) {
-            throw new IllegalArgumentException(formattedErrorMessage("orgUrl.invalid", url), e);
-        }
+    public static ValidationResponse validateOrgUrl(String url) {
+        return validateHttpsUrl(url, "orgUrl");
     }
 
     /**
      * Validates that an API token is not null and contains does not contain the string {@code {apiToken}}.
      * @param token The API Token to be validated
      */
-    public static void validateApiToken(String token) {
-        hasText(token, ERRORS.getString("apiToken.missing"));
-        doesNotContain(token, "{apiToken}",  ERRORS.getString("apiToken.invalid"));
+    public static void assertApiToken(String token) {
+        validateApiToken(token).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateApiToken(String token) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(token)) {
+            response.setMessage(ERRORS.getString("apiToken.missing"));
+        } else if (contains(token, "{apiToken}")) {
```

Comment:
Are you just looking for brackets or for that whole string?

Reply:
could make it a regex, if you want to punish yourself ;) `{[^{}]*}`

Reply:
Just that specific string, there are some edge cases where customers can define custom strings for clientId's (though i'm guessing it isn't true for tokens)